### PR TITLE
Add new core configuration UI for external_url & internal_url

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

Add a UI for the new `external_url` and `internal_url` core configuration options.

image

When configured via YAML:

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/13307
- Link to parent pull request: https://github.com/home-assistant/core/pull/35224
- Link to JavaScript WebSocket pull request: https://github.com/home-assistant/home-assistant-js-websocket/pull/115

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/core/ha-config-url-form.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+  private async _save() {
+    this._working = true;
+    try {
+      await saveCoreConfig(this.hass, {
+        external_url: this._external_url || null,
+        internal_url: this._internal_url || null,
+      });
+    } catch (err) {
+      alert("Invalid URLs");
```
Comment (bramkragten) [inline src/panels/config/core/ha-config-url-form.ts:128]: Don't show an alert but save the error you get from the backend in `this._error` and print that at the top of the form in the `error-color`. 
```ts
${this._error ? html`<div class="error">${this._error}</div>` : ""}
```

Comment (bramkragten) [reply src/panels/config/core/ha-config-url-form.ts:128]: And reset the error when you start saving.

Comment (frenck) [reply src/panels/config/core/ha-config-url-form.ts]: Done!

File: src/panels/config/core/ha-config-url-form.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+class ConfigUrlForm extends LitElement {
+  @property() public hass!: HomeAssistant;
+
+  @property() private _error?: string;
+
+  @property() private _working = false;
+
+  @property() private _external_url!: string;
+
+  @property() private _internal_url!: string;
```
Comment (bramkragten) [inline src/panels/config/core/ha-config-url-form.ts:28]: This is undefined on start right? So the `!` should be `?`

Same for external.

**Review by bramkragten — APPROVED** [review]
Ok to merge after comment

File: src/panels/config/core/ha-config-url-form.ts [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+  protected render(): TemplateResult {
+    const canEdit = ["storage", "default"].includes(
+      this.hass.config.config_source
+    );
+    const disabled = this._working || !canEdit;
+
+    return html`
+      ${this.hass.userData?.showAdvanced
```
Comment (bramkragten) [inline src/panels/config/core/ha-config-url-form.ts:37]: This check should be on top of render outside the template:
```
If(!advanced) {
 return html``
}
```

Comment (frenck) [reply src/panels/config/core/ha-config-url-form.ts]: Right add a guard

**Comment by frenck** [comment]
Rebased the PR to resolved the merge conflicts.

**Review by balloob — APPROVED** [review]

**Pull request merged**